spec/jobs/get_usps_proofing_results_job_spec.rb

```diff
@@ -243,34 +243,89 @@
 
-      it 'logs a message with counts of various outcomes when the job completes' do
-        allow(InPersonEnrollment).to receive(:needs_usps_status_check).
-          and_return(pending_enrollments)
-        stub_request_proofing_results_with_responses(
-          request_passed_proofing_results_args,
-          request_in_progress_proofing_results_args,
-          { status: 500 },
-          request_failed_proofing_results_args,
-          request_expired_proofing_results_args,
-        )
+      context 'with a set of outcomes to log' do
+        let(:responses) do
+          [
+            request_passed_proofing_results_args,
+            request_in_progress_proofing_results_args,
+            { status: 500 },
+            request_failed_proofing_results_args,
+            request_expired_proofing_results_args,
+          ]
+        end
+
+        before do
+          allow(InPersonEnrollment).to receive(:needs_usps_status_check).
+            and_return(pending_enrollments)
+          stub_request_proofing_results_with_responses(*responses)
+        end
 
-        job.perform(Time.zone.now)
+        it 'logs a message with counts of various outcomes when the job completes' do
+          job.perform(Time.zone.now)
 
-        expect(job_analytics).to have_logged_event(
-          'GetUspsProofingResultsJob: Job completed',
-          duration_seconds: anything,
-          enrollments_checked: 5,
-          enrollments_errored: 1,
-          enrollments_expired: 1,
-          enrollments_failed: 1,
-          enrollments_in_progress: 1,
-          enrollments_passed: 1,
-        )
+          expect(job_analytics).to have_logged_event(
+            'GetUspsProofingResultsJob: Job completed',
+            duration_seconds: anything,
+            enrollments_checked: 5,
+            enrollments_errored: 1,
+            enrollments_expired: 1,
+            enrollments_failed: 1,
+            enrollments_in_progress: 1,
+            enrollments_passed: 1,
+          )
 
-        expect(
-          job_analytics.events['GetUspsProofingResultsJob: Job completed'].
-            first[:duration_seconds],
-        ).to be >= 0.0
+          expect(
+            job_analytics.events['GetUspsProofingResultsJob: Job completed'].
+              first[:duration_seconds],
+          ).to be >= 0.0
+        end
+
+        context 'when the job is terminated with a ConcurrencyExceededError' do
+          let(:analytics_title) { 'GetUspsProofingResultsJob: Job terminated' }
+          let(:error_message) { 'too many!' }
+          let(:error_job_class) do
+            Class.new(GetUspsProofingResultsJob) do
+              private
+
+              def check_enrollment(enrollment, is_not_last_enrollment)
+                if is_not_last_enrollment
+                  super(enrollment, is_not_last_enrollment)
+                else
+                  raise GoodJob::ActiveJobExtensions::Concurrency::ConcurrencyExceededError,
+                        'too many!'
```

Comment:
```suggestion
                        error_message
```

Reply:
I think some weird scoping rules would prevent that unforunately, we'd have to capture `error_message` and then use the `define_method` syntax

Reply:
yeah, Zach is right, scoping prevents using our variables here.